src/realm/object-store/sectioned_results.cpp

```diff
@@ -216,40 +216,103 @@ struct SectionedResultsNotificationHandler {
     // When set change notifications will be filtered to only deliver
     // change indices refering to the supplied section key.
     util::Optional<Mixed> m_section_filter;
-    // Should be set when the key is StringData or BinaryData.
-    util::Optional<std::shared_ptr<std::string>> m_section_key_buffer;
 };
 
+template <typename StringType>
+void create_buffered_key(Mixed& key, util::Optional<std::string>& buffer, StringType value)
+{
+    if (value.size() == 0) {
+        key = StringType("", 0);
+    }
+    else {
+        buffer = std::string(value.data(), value.size());
+        key = Mixed(buffer);
+    }
+}
+
+ResultsSection::ResultsSection()
+    : m_parent(nullptr)
+    , m_key(Mixed())
+    , m_key_buffer(util::none)
+{
+}
+
+ResultsSection& ResultsSection::operator=(ResultsSection&& other)
```

Comment:
Anything which defines `operator=()` also needs to define the move constructor (and vice versa). Storing the buffer in a `std::unique_ptr<char[]>` rather than an optional `std::string` would make the defaulted version work for both, as it means that the buffer stays in the same place in memory when moving a ResultsSection.

Reply:
I've removed the non default `operator=()` and replaced the `util::Optional<std::string>` with `std::unique_ptr<char[]>`